### PR TITLE
[MOB-10032] Example App: Login page

### PR DESCRIPTION
**Pull request by lposen**
## 🔹 JIRA Ticket(s) if any

* [MOB-10032](https://iterable.atlassian.net/browse/MOB-10032)

## ✏️ Description

Added an example app -- currently, it only has a login page.  More will follow.

## Testing

- cd into the root directory
- run `yarn install`
- cd into the example directory
- follow the instructions for setup in the example directories readme

### iOS
Simulator Screenshot - iPhone 11 - 2024-11-06 at 15 55 23
Simulator Screenshot - iPhone 11 - 2024-11-06 at 15 55 34

### Android
Screenshot 2024-11-06 at 4 01 22 PM
Screenshot 2024-11-06 at 4 01 44 PM

[MOB-10032]: https://iterable.atlassian.net/browse/MOB-10032?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by evantk91 — APPROVED** [review]
Looks amazing. Just a few clarifications.

File: example/.env.example [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# This file contains environment variables that are used by the example scripts.
+# To use the example scripts, you must first copy this file to `.env` and
+# replace the placeholder text with your own values.
+
+# To add an API key, do the following:
+#  1. Sign into your Iterable account
+#  2. Go to [Integrations > API Keys](https://app.iterable.com/settings/apiKeys)
+#  3. Click "New API Key" in the top right corner
+#  4. Fill in the followsing fields:
```
[the remark this answers is not in the document]
Comment (lposen) [reply example/.env.example:9]: Thanks!   Gonna push a fix soon.

File: example/src/constants/styles/colors.ts [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+export const colors = {
+  ...baseColors,
+  ...brandColors,
+  ...utilityColors,
+  ...textColors,
+  ...backgroundColors,
+  ...linkColors,
+  ...borderColors,
+  ...iconColors,
```
[the remark this answers is not in the document]
Comment (lposen) [reply example/src/constants/styles/colors.ts:318]: iterable-frontend: https://github.com/Iterable/iterable-frontend/blob/main/src/ui/theme/colors.ts

File: example/src/hooks/useIterableApp.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        setApiKey,
+        setLoginInProgress,
+        setUserId,
+        userId,
+      }}
+    >
+      {children}
+    </IterableAppContext.Provider>
+  );
+};
```
[the remark this answers is not in the document]
Comment (lposen) [reply example/src/hooks/useIterableApp.tsx:208]: Yeah -- it's a great way to keep common functionality together.  Would love to implement something similar in our actual app.

File: example/src/hooks/useIterableApp.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+const EMAIL_REGEX = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+export const IterableAppProvider: FunctionComponent<
+  React.PropsWithChildren<unknown>
+> = ({ children }) => {
+  const [itblConfig, setItblConfig] = useState<IterableConfig | null>(null);
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
+  const [isInitialized, setIsInitialized] = useState<boolean>(false);
+  const [apiKey, setApiKey] = useState<string | undefined>(
+    process.env.ITBL_API_KEY
```
[the remark this answers is not in the document]
Comment (lposen) [reply example/src/hooks/useIterableApp.tsx:82]: Thanks!

**Pull request merged**